File: src/routes/health.ts
import { Router, Request, Response } from 'express';
import { ApiResponse } from '../types/global';

const router = Router();

router.get('/healthz', (req: Request, res: Response): void => {
  const response: ApiResponse<{ status: string; uptime: number; timestamp: string }> = {
    success: true,
    data: {
      status: 'healthy',
      uptime: process.uptime(),
      timestamp: new Date().toISOString(),
    },
    timestamp: new Date().toISOString(),
  };

  res.json(response);
});

router.get('/readyz', (req: Request, res: Response): void => {
  // In a real app, you'd check database connections, external services, etc.
  const checks = {
    environment: !!process.env.NODE_ENV,
    apiKey: !!process.env.API_KEY_PARTNER,
    openai: !!process.env.OPENAI_API_KEY,
    plaid: !!process.env.PLAID_CLIENT_ID && !!process.env.PLAID_SECRET,
  };

  const allReady = Object.values(checks).every(Boolean);

  const response: ApiResponse<{ ready: boolean; checks: typeof checks }> = {
    success: true,
    data: {
      ready: allReady,
      checks,
    },
    timestamp: new Date().toISOString(),
  };

  res.status(allReady ? 200 : 503).json(response);
});

export default router;

File: app/routes/health.py
"""Health check endpoints."""

import time
from typing import Dict, Any

from fastapi import APIRouter, status
from pydantic import BaseModel

from app.core.config import settings


class HealthResponse(BaseModel):
    """Health check response model."""
    
    status: str
    timestamp: float
    version: str = "1.0.0"


class ReadinessResponse(BaseModel):
    """Readiness check response model."""
    
    status: str
    timestamp: float
    version: str = "1.0.0"
    checks: Dict[str, Any]


router = APIRouter()


@router.get(
    "/healthz",
    response_model=HealthResponse,
    status_code=status.HTTP_200_OK,
    summary="Health check",
    description="Basic health check endpoint",
)
async def health_check() -> HealthResponse:
    """Health check endpoint."""
    return HealthResponse(
        status="healthy",
        timestamp=time.time(),
    )


@router.get(
    "/readyz",
    response_model=ReadinessResponse,
    status_code=status.HTTP_200_OK,
    summary="Readiness check",
    description="Readiness check endpoint with dependency validation",
)
async def readiness_check() -> ReadinessResponse:
    """Readiness check endpoint."""
    checks = {}
    
    # Check critical environment variables
    checks["config"] = "ok"
    
    # Check OpenAI configuration
    checks["openai"] = "ok" if settings.OPENAI_API_KEY else "missing_key"
    
    # Check Plaid configuration
    checks["plaid"] = "ok" if settings.PLAID_CLIENT_ID and settings.PLAID_SECRET else "missing_credentials"
    
    # Overall status
    overall_status = "ready" if all(check == "ok" for check in checks.values()) else "not_ready"
    
    return ReadinessResponse(
        status=overall_status,
        timestamp=time.time(),
        checks=checks,
    )